.github/workflows/formatting.yaml

```diff
@@ -3,15 +3,24 @@ name: OpenTurbine-CI
 on: push
 
 jobs:
-  ClangFormat:
+  clang-format:
     runs-on: ubuntu-latest
+
     steps:
-    - name: Clone
+    - name: Checkout repository
       uses: actions/checkout@v4
-    - name: Check formatting
-      uses: DoozyX/clang-format-lint-action@v0.16.2
-      with:
-        source: './src ./tests/unit_tests'
-        exclude: '.'
-        extensions: 'cpp,hpp'
-        clangFormatVersion: 16
+
+    - name: Run clang-format
+      run: |
+        find . -regex '.*\.\(cpp\|hpp\|c\|h\)' -exec clang-format -i {} \;
+        
+    - name: Check for formatting changes
+      run: |
+        if [[ `git status --porcelain` ]]; then
+          echo "Code is not formatted. Please run clang-format."
```

Comment:
Would the following message make things a little clearer?

`echo "::error::Code is not properly formatted. Please run clang-format locally and commit the changes."`

Reply:
Maybe.  Honestly, I don't think anybody will read this message - if there's a formatting error, you'll get a big red X and the dozens of lines of git-diff.